app/src/services/storage.js

```diff
@@ -79,163 +135,188 @@ const objectStorageService = {
       }).join('&');
     }
 
-    return this._s3Client.send(new CopyObjectCommand(params));
+    return this._getS3Client(data).send(new CopyObjectCommand(params));
   },
 
   /**
    * @function deleteObject
    * Deletes the object at `filePath`
    * @param {string} options.filePath The filePath of the object
-   * @param {number} [options.versionId = undefined] Optional specific versionId for the object
+   * @param {number} [options.versionId] Optional specific versionId for the object
+   * @param {string} [options.bucketId] Optional bucketId
    * @returns {Promise<object>} The response of the delete object operation
    */
-  deleteObject({ filePath, versionId = undefined }) {
+  async deleteObject({ filePath, versionId = undefined, bucketId = undefined }) {
```

Comment:
just me thinking again.. oh no.. so when we call these from these storage methods, we want to pass in the bucketId from the object controller, i wonder if the original API request needs to have a bucketId input parameter.. or whether we can look that up in the db using the bucketId column.

Reply:
The service layer does not care about how the bucketId is procured as that's the controller's perogative. That being said, the general intention for a deleteObject call from the API is that we determine the bucketId by inherently polling the object.bucketId column. This will likely be wired in from req.currentObject as we already populate bucketId by definition there from the middleware operations, so the controller does not need to invoke any extra DB calls to proceed.

Reply:
ok.. so that would be a reason to keep currentObject separate from hasPermissions.